src/main/java/request/HttpRequestReader.java

```diff
@@ -0,0 +1,13 @@
+package com.td.HttpServer;
+import java.io.IOException;
+
+public class HttpRequestReader implements IRequestReader {
```

Comment:
Feels like this class does nothing more than delegate to IMessageClient. What's the purpose for it?